Libraries/Dijkstra/Dijkstra.h

```diff
@@ -0,0 +1,262 @@
+        matrix[current_coord->x][current_coord->y - 1].prev = matrix[current_coord->x][current_coord->y].current;
+      }
+    }
+    open_vector.popFront();
+    
+  }
+
+  if (non_visited_vector.getSize() > 0) current_coord = non_visited_vector[0];
+  else Serial.print("No hay tiles por recorrer");
+
+  //Comparing all the non Visited Tiles
+  for (int i = 1 ; i < non_visited_vector.getSize() ; ++i) {
+    if (matrix[current_coord->x][current_coord->y].weight > matrix[non_visited_vector[i]->x][non_visited_vector[i]->y].weight) {
+      current_coord = non_visited_vector[i];
+    }
+  }
+
+  ///////////////////////////////////////////////////////////////////////////////
+  for (int i = 0 ; i < tiles_map.getRows() ; ++i) {
+    for (int j = 0 ; j < tiles_map.getColumns() ; ++j) {
+      Serial.print("(");
+      Serial.print(matrix[i][j].current.x);
+      Serial.print(",");
+      Serial.print(matrix[i][j].current.y);
+      Serial.print(")");
+      Serial.print("[");
+      Serial.print(matrix[i][j].prev.x);
+      Serial.print(",");
+      Serial.print(matrix[i][j].prev.y);
+      Serial.print("]");
+      Serial.print(matrix[i][j].weight);
+      Serial.print("  ");
+    }
+    Serial.println("");
+  }
+  Serial.println("");
+  ////////////////////////////////////////////////////////////////////////////////
+
+  while (current_coord->x != current_row || current_coord->y != current_column) {
+    if (current_coord->x - matrix[current_coord->x][current_coord->y].prev.x == -1) {
+      path.pushFront('N');
+    }
+    if (current_coord->y - matrix[current_coord->x][current_coord->y].prev.y == 1) {
+      path.pushFront('E');
+    }
+    if (current_coord->x - matrix[current_coord->x][current_coord->y].prev.x == 1) {
+      path.pushFront('S');
+    }
+    if (current_coord->y - matrix[current_coord->x][current_coord->y].prev.y == -1) {
+      path.pushFront('W');
+    }
+    Serial.print("(");
+    Serial.print(current_coord->x);
+    Serial.print(",");
+    Serial.print(current_coord->y);
+    Serial.print(")");
+    Serial.print("  ");
+    current_coord = &matrix[current_coord->x][current_coord->y].prev;
+  }
+    Serial.print(current_coord->x);
+    Serial.print(",");
+    Serial.print(current_coord->y);
+    Serial.print("  ");
+
+  return path;
+}
+
+TVector<char> goToStart(Map tiles_map) {
+  TVector<coord*> open_vector;
+  TVector<char> path;
+  dijkstra_tile matrix[tiles_map.getRows()][tiles_map.getColumns()];
+  coord *current_coord;
+  uint8_t current_row = tiles_map.getCurrRow();
+  uint8_t current_column = tiles_map.getCurrColumn();
+
+  for (int i = 0 ; i < tiles_map.getRows() ; ++i) {
+    for (int j = 0 ; j < tiles_map.getColumns() ; ++j) {
+      matrix[i][j].current.x = i;
+      matrix[i][j].current.y = j;
+      matrix[i][j].weight = 255;
+      matrix[i][j].checked = false;
+    }
+  }
```

Comment:
This is repeated in Dijkstra, make it a function.

Reply:
I have been having trouble with passing the matrix as a parameter without making it a global variable. Same as in the previous comment.